src/coreclr/pal/src/init/pal.cpp

```diff
@@ -288,6 +288,14 @@ InitializeDefaultStackSize()
         }
     }
 
+#ifdef HOST_OSX
```

Comment:
I probably also need `HOST_MACCATALYST` here and the other spot :)

Reply:
macCatalyst is not really a desktop platform. Also, regular CoreCLR does not work on macCatalyst.

Reply:
Oh well, that's a shame. I feel like it's a semi-desktop platform, since it clearly is meant to run on a desktop OS. Would we not want to make this guarantee for .NET MAUI on macOS for example (which doesn't support real macOS last time I checked, only catalyst)?
I could change the test projects to also run on maccatalyst, or should I just leave them as-is?

Reply:
macCatalyst is a flavor of iOS. For example, `OperatingSystem.IsiOS` returns true for macCatalyst, the mac Catalyst apps have to conform with similar requirements as iOS apps, JITing is not allowed in macCatalyst apps, ... .

I do not think it makes sense to have different settings for macCatalyst and other Apple device platforms.

Reply:
> macCatalyst is a flavor of iOS

Yes, but it is a flavour which is clearly intended to run on macOS devices (a desktop OS).

We already have a number of APIs which are supported on mac catalyst, but not iOS:
- https://github.com/dotnet/runtime/blob/5d3a3c1e5d3185e1e1b169e0a3479326963f4d69/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.OSX.cs#L20
- https://github.com/dotnet/runtime/blob/5d3a3c1e5d3185e1e1b169e0a3479326963f4d69/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessThread.iOS.cs#L27
- https://github.com/dotnet/runtime/blob/5d3a3c1e5d3185e1e1b169e0a3479326963f4d69/src/libraries/System.IO.Ports/Directory.Build.props#L5
- https://github.com/dotnet/runtime/blob/5d3a3c1e5d3185e1e1b169e0a3479326963f4d69/src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFilesWatcher.cs#L92
- https://github.com/dotnet/runtime/blob/5d3a3c1e5d3185e1e1b169e0a3479326963f4d69/src/libraries/System.Private.CoreLib/src/System/Environment.iOS.cs#L14

I don't think it's such an issue for the setting to be different on macCatalyst from iOS, given it's intended to run on a desktop OS.

Reply:
@vitek-karas @kotlarmilos @rolfbjarne This PR is changing the default stack size on secondary threads on macOS from 512kB to 1.5MB for consistency with other platforms

Do you have an opinion on whether we should make the same change macCatalyst? (It would make it different from the iOS default.)

Reply:
Note that mono is already 2MB on 64-bit platforms we have discovered. So, this would only affect any other platforms it can (or will in the future) run on, e.g., NativeAOT.

Reply:
Memory is never swapped out on iOS, while on Mac Catalyst it is, which is a big difference.

Also I wouldn't expect allocated memory to be all that significant in terms of actual memory usage until the memory in question is read/modified.

So I think it would make sense to make Mac Catalyst behave like a desktop platform in this case.

Reply:
@ivanpovazan for additional opinion

Reply:
Sorry for my later reply and thanks @hamarb123 for looking into this.

I agree with what @jkotas proposed of having the changed unified for all Apple platforms.
Additionally, `+1` on having this properly documented in regard to the supported platforms and differences between the runtimes (Mono vs NativeAOT vs CoreCLR).